src/OpenTelemetry.Api/Trace/SpanContextNew.cs

```diff
@@ -0,0 +1,165 @@
+﻿// <copyright file="SpanContextNew.cs" company="OpenTelemetry Authors">
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+// </copyright>
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Linq;
+using OpenTelemetry.Api.Context.Propagation;
+
+namespace OpenTelemetry.Trace
+{
+    /// <summary>
+    /// A class that represents a span context. A span context contains the state that must propagate to
+    /// child <see cref="TelemetrySpan"/> and across process boundaries. It contains the identifiers <see cref="ActivityTraceId"/>
+    /// and <see cref="ActivitySpanId"/> associated with the <see cref="TelemetrySpan"/> and a set of <see cref="TraceFlags"/>.
+    /// </summary>
+    public readonly struct SpanContextNew
+    {
+        /// <summary>
+        /// A blank <see cref="SpanContextNew"/> that can be used for remote no-op operations.
+        /// </summary>
+        internal static readonly SpanContextNew BlankRemote = new SpanContextNew(default, default, ActivityTraceFlags.None, true, null);
+
+        internal readonly ActivityContext ActivityContext;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="SpanContextNew"/> struct with the given identifiers and options.
+        /// </summary>
+        /// <param name="traceId">The <see cref="ActivityTraceId"/> to associate with the <see cref="SpanContextNew"/>.</param>
+        /// <param name="spanId">The <see cref="ActivitySpanId"/> to associate with the <see cref="SpanContextNew"/>.</param>
+        /// <param name="traceFlags">The <see cref="TraceFlags"/> to
+        /// associate with the <see cref="SpanContextNew"/>.</param>
+        /// <param name="isRemote">The value indicating whether this <see cref="SpanContextNew"/> was propagated from the remote parent.</param>
+        /// <param name="tracestate">The tracestate to associate with the <see cref="SpanContextNew"/>.</param>
+        public SpanContextNew(in ActivityTraceId traceId, in ActivitySpanId spanId, ActivityTraceFlags traceFlags, bool isRemote = false, IEnumerable<KeyValuePair<string, string>> tracestate = null)
+        {
+            this.ActivityContext = new ActivityContext(traceId, spanId, traceFlags, TracestateUtils.GetString(tracestate));
```

Comment:
@cijothomas I understand that you picked `Tracestate` rather than `TraceState` since the W3C TraceContext spec treats tracestate as a single word. I think folks started to use `TraceState` as `Tracestate` is very counter intuitive and many editors/IDEs would complain about it. Further more, .NET API is using `TraceState`.

Net net, I think we should use `TraceParent` and `TraceState` as variable names consistently across this repo. The W3C spec has said nothing about what the variable name should be for a programming language.

* https://docs.microsoft.com/en-us/dotnet/api/system.diagnostics.activitycontext.tracestate?view=net-5.0
* https://github.com/open-telemetry/opentelemetry-dotnet/blob/30f841b2eff8e47a3e680a7afbea3ab528faa5d2/src/OpenTelemetry.Api/Context/Propagation/TraceContextFormat.cs#L31
* https://github.com/open-telemetry/opentelemetry-python/blob/9d7491845064dfb5d8f305cbb1d5cc1d479fe8fd/opentelemetry-api/src/opentelemetry/trace/span.py#L140
* https://github.com/open-telemetry/opentelemetry-java/blob/e1ddcaec24eedbd8ffc5f2355249d441db9f5af0/api/src/main/java/io/opentelemetry/trace/TraceState.java

Reply:
Agree.